### DISCUSSION TITLE
Umi & Bigfish 想法征集 07

### DISCUSSION THREAD

**Discussion by sorrycc** [umijs/umi#10873] [Ideas]
> Why this issue? 计划在 Umi Contributor 群增加一条规则。每人每周贡献 2 个改进 Umi 或 Bigfish 的想法（辟起、辟殊、云谦），外包同学减半（1 个），外部 Contributor 随意，我每周三会在群里发个 issue 做收集。

直接回复即可。（含内部敏感信息的可以私发我）

**Comment by sorrycc** [comment]
- 整理各种 SOP（标准作业流程）流程，比如发布、答疑、开发插件、写各种类型的代码，让任何同学都可基于 SOP 流程完成操作。SOP 之后再考虑自动化（RPA）。
- 框架里 format 的部分改用 rome 代替 prettier 。
- 期贤整了个基于 chatgpt 的 dva to hooks 代码改写工具。

**Reply by stormslowly** [reply]
第 2 点，我在想是不是也要把 e2e 换成默认快的实现。 @fz6m

**Reply by fz6m** [reply]
e2e 建议换 playwright

**Reply by stormslowly** [reply]
需要一个地方托管下 用例的视频~

**Comment by stormslowly** [comment]
1. 在 dev 命令下，如果用户写了一段预发错误的代码，prepare 阶段的 esbuild 报错会让 dev 进程退出，DX 体验比较差；在 dev 应该不退出。
2. 目前对 antd 的 ConfigProvider 有两个插件 antd 和 locale 可注入，如果两个插件同时注入，antd 插件中的配置就会被覆盖掉，所以这两个插件需要协调好 config provider 的值设置，初步想法是 locale 开启 antd 就需要修改 antd.configProvider , 同时 local 通过 antd 运行时配置注入。( @hanzebang  这个可以放 doctor 如果用户用了两个 Antd 的 ConfigProvider 可以给一个 warning )
3.  如果用户使用了 tech-ui, 是否应该提示用户包裹一个 ProProvider 

（ 这周和 Provider 杠上了 。。。

**Reply by hanzebang** [reply]
关于问题2，我理解：
1、ConfigProvider 多层嵌套是支持的，应该并没有什么问题，除非他俩设置了相同的属性
2、我记着他俩的角度应该是不同的，locale 的 ConfigProvider 应该仅会设置 locale 属性，而 antd 的 ConfigProvider 一般是设置其他属性

**Comment by PeachScript** [comment]
- 幽灵依赖检查需要支持 `optionalDependencies`，涉及 umi、father、bigfish-doctor
- Bigfish 讨论区可能需要个机器人推送新帖子
- jest 编译到 esm 的时候如果能有明确的错误提示引导配置 `transformIgnorePatterns` 会降低一些答疑量

**Reply by fz6m** [reply]
jest 的 esm 有很多限制，esbuild `0.16.x` 内部加了限制，不允许 jest hack esm 了，运行会报错。

建议后续用纯 native esm 的 vitest 来测试项目，jest 仅用于测试纯 cjs 产物的 node 逻辑。
